packages/core/votes/sequencetracker.go

```diff
@@ -116,14 +116,17 @@ func (s *SequenceTracker[VotePowerType]) addVoteToMarker(marker markers.Marker,
 		return
 	}
 
-	// Trigger events for all newly supported markers.
-	for i := previousHighestIndex + 1; i <= marker.Index(); i++ {
-		s.Events.VoterAdded.Trigger(&SequenceVoterEvent{
-			Voter:  voter,
-			Marker: markers.NewMarker(marker.SequenceID(), i),
-		})
+	if previousHighestIndex == 0 {
+		sequence, _ := s.sequenceCallback(marker.SequenceID())
+		previousHighestIndex = sequence.LowestIndex()
 	}
 
+	s.Events.SequenceVotersUpdated.Trigger(&SequenceVotersUpdatedEvent{
```

Comment:
The intention is to trigger an event for each updated sequence right? But can it happen that there's somehow multiple events per sequence?

Reply:
If sequence A references sequence B, and then sequence B references sequence A? 
BTW. should we also have a DAG mutex on markers in acceptance gadget in case two voters update the same marker?